spdm.rflx

```diff
@@ -0,0 +1,1506 @@
+            then Req_Alg_Structs
+               --  Length is the total length of the message, we have to subtract the common header
+               --  of Request.
+               with Size => 8 * (Length_2 * 256 + Length_1) - 16 - (Ext_Hash'Last - Req_Alg_Struct_Count'First + 1)
+                  if 8 * (Length_2 * 256 + Length_1) >= 16 + (Ext_Hash'Last - Req_Alg_Struct_Count'First + 1);
+         Req_Alg_Structs : Req_Alg_Structs;
+         -- ISSUE: Componolit/RecordFlux#51
+         -- The number of elements should also match Req_Alg_Struct_Count
+      end message;
+
+   type Resp_Alg_Structs is sequence of Alg_Struct;
+
+   --  ISSUE: Componolit/RecordFlux#104
+   --  This should be little endian
+   type Measurement_Hash_Algo is
+      (Raw_Bit_Streams_Only => 2147483648,
+       MH_TPM_ALG_SHA_256   => 1073741824,
+       MH_TPM_ALG_SHA_384   => 536870912,
+       MH_TPM_ALG_SHA_512   => 268435456,
+       MH_TPM_ALG_SHA3_256  => 134217728,
+       MH_TPM_ALG_SHA3_384  => 67108864,
+       MH_TPM_ALG_SHA3_512  => 33554432)
+   with Size => 32;
+
+   --  ISSUE: Componolit/RecordFlux#104
+   --  This should be little endian
+   type Base_Asym_Sel is
+      (BA_Unsupported                 => 2147483648,
+       BA_TPM_ALG_RSASSA_2048         => 1073741824,
+       BA_TPM_ALG_RSAPSS_2048         => 536870912,
+       BA_TPM_ALG_RSASSA_3072         => 268435456,
+       BA_TPM_ALG_RSAPSS_3072         => 134217728,
+       BA_TPM_ALG_ECDSA_ECC_NIST_P256 => 67108864,
+       BA_TPM_ALG_RSASSA_4096         => 33554432,
+       BA_TPM_ALG_RSAPSS_4096         => 16777216,
+       BA_TPM_ALG_ECDSA_ECC_NIST_P384 => 8388608,
+       BA_TPM_ALG_ECDSA_ECC_NIST_P521 => 4194304)
+   with Size => 32;
+
+   --  ISSUE: Componolit/RecordFlux#104
+   --  This should be little endian
+   type Base_Hash_Sel is
+      (BH_Unsupported      => 2147483648,
+       BH_TPM_ALG_SHA_256  => 1073741824,
+       BH_TPM_ALG_SHA_384  => 536870912,
+       BH_TPM_ALG_SHA_512  => 268435456,
+       BH_TPM_ALG_SHA3_256 => 134217728,
+       BH_TPM_ALG_SHA3_384 => 67108864,
+       BH_TPM_ALG_SHA3_512 => 33554432)
+   with Size => 32;
+
+   type Algorithms_Response is
+      message
+         Resp_Alg_Struct_Count : Alg_Struct_Count;
+         Param_2 : Reserved_8;
+         Length_1 : Length;
+         Length_2 : Length
+            then Measurement_Specification_Sel
+               if Length_2 * 256 + Length_1 <= 128;
+         Measurement_Specification_Sel: Measurement_Specification;
+         Reserved_1 : Reserved_8;
+         Measurement_Hash_Algo : Measurement_Hash_Algo;
+         Base_Asym_Sel : Base_Asym_Sel;
+         Base_Hash_Sel : Base_Hash_Sel;
+         Reserved_2 : Reserved_64;
+         Reserved_3 : Reserved_32;
+         Ext_Asym_Count : Ext_Asym_Count
+            then Ext_Hash_Count
+               if Ext_Asym_Count <= 1;
+         Ext_Hash_Count : Ext_Hash_Count
+            then Reserved_4
+               if Ext_Hash_Count <= 1;
```

Comment:
Called `ExtAsymSelCount` and `ExtHashSelCount` in specification.

BTW: Field conditions could be used here (instead of link conditions, i.e., keeping the `if` and just removing the `then X` would lead to a semantically equivalent model).